runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -115,7 +115,13 @@ def dataflowFnApiWorkerJar = project.findProperty('dataflowWorkerJar') ?: projec
 def dockerImageRoot = project.findProperty('dockerImageRoot') ?: "us.gcr.io/${dataflowProject}/java-postcommit-it"
 def dockerImageContainer = "${dockerImageRoot}/java"
 def dockerTag = new Date().format('yyyyMMddHHmmss')
-def dockerImageName = "${dockerImageContainer}:${dockerTag}"
+ext.dockerImageName = "${dockerImageContainer}:${dockerTag}"
```

Comment:
I'm not sure whether it would be better to share this attr, but the default docker container for postcommit is "us.gcr.io/${dataflowProject}/java-postcommit-it", and the default docker container for precommit is "us.gcr.io/${gcpProject}/java-examples-it". I did this in order to reduce the possibility of race condition during pushing and deleting images.

Reply:
Oh, interesting. I had assumed that these tests were using the same Docker container and so we could just build and upload them once in a shared task.

Are there real differences in the container images such that they should be kept separate?

Reply:
So sorry for the late. There is no real differences if launching all tests at the same time. The reason why I do this here is, the post commit and pre commit could be invoked at the same time, then we may have the race condition for the same image name. We can discuss offline if preferred.

Reply:
With this refactoring, both test suites (precommit, postcommit) share a single `buildAndPushDockerContainer` task, so if both test suites were to execute together, `buildAndPushDockerContainer` would only be executed once and the containers re-used for both suites. I believe this is preferable.

Reply:
I feel like, even though they share the same task definition, buildAndPushDockerContainer will still be invoked per test task. For race condition, imaging there are 2 Jenkins machines running precommit and postcommit at the same time, then they will try to push the same docker image into the same gcr repo, which may cause some problems.

Reply:
Valid point. The `buildAndPushDockerContainer` task will only run once per gradle execution task graph, so if a single jenkins job ran both pre-commits and post-commits they would share a single container image on gcr. But you're right that concurrent builds from different jenkins machines could possibly overwrite each other.

The current tag scheme includes a timestamp up to second precision, so a collision is possible but unlikely. This risk of races between concurrent builds is an existing possibility even without this change. 

Do you think we need to address it with this PR or can it be deferred?